horace_core/Tobyfit/instpars_DGfermi.m

```diff
@@ -31,32 +31,58 @@
 % Get array of instruments
 instruments = header.instruments;
 nrun=instruments.n_runs;
-inst=repmat({instruments{1}},[nrun,1]);
-for i=2:nrun
-    inst{i}=instruments{i};
-end
 
-% Fill output arguments
 ei=zeros(nrun,1);
 x0=zeros(nrun,1);
 xa=zeros(nrun,1);
 x1=zeros(nrun,1);
 thetam=zeros(nrun,1);
 angvel=zeros(nrun,1);
-moderator=repmat(IX_moderator,[nrun,1]);
-aperture=repmat(IX_aperture,[nrun,1]);
-chopper=repmat(IX_fermi_chopper,[nrun,1]);
 for i=1:nrun
     ei(i)=header.expdata(i).efix;
-    x1(i)=abs(inst{i}.fermi_chopper.distance);
-    x0(i)=abs(inst{i}.moderator.distance) - x1(i);      % distance from Fermi chopper to moderator face
-    xa(i)=abs(inst{i}.aperture.distance) - x1(i);       % distance from Fermi chopper to beam defining aperture
-    thetam(i)=inst{i}.moderator.angle*(pi/180);         % angle of moderator face to inc. beam (radians)
-    angvel(i)=inst{i}.fermi_chopper.frequency*(2*pi);   % angular velocity of Fermi chopper
-    moderator(i)=inst{i}.moderator;
-    aperture(i)=inst{i}.aperture;
-    chopper(i)=inst{i}.fermi_chopper;
-    chopper(i).energy=ei(i);                            % Update incident energy to value in header
+end
+
+use_unique_objects = true;
+% switch between old and new inputs to object lookup
+% the old version will be removed after testing
+if ~use_unique_objects
+
+	inst=repmat({instruments{1}},[nrun,1]);
+	for i=2:nrun
+	    inst{i}=instruments{i};
+	end
```

Comment:
What's the difference between this and 
```
inst = instruments;
```

Reply:
this is the original unmodified code. It has just been rearranged slightly to accommodate the if/else. In due time it will vanish when we no longer need the easy switch to the old version